scripts/generate_sparse_test_matrix.ts

```diff
@@ -0,0 +1,31 @@
+import { SparseTestMatrixGenerator } from './components/sparse_test_matrix_generator.js';
+
+// This script generates a sparse test matrix.
+// Every test must run on each type of OS and each version of node.
+// However, we don't have to run every combination.
+
+if (process.argv.length < 3) {
+  console.log(
+    "Usage: npx tsx scripts/generate_sparse_test_matrix.ts '<test-glob-pattern>' <max-tests-per-job?>"
+  );
+}
+
+const testGlobPattern = process.argv[2];
+const maxTestsPerJob = process.argv[3] ? parseInt(process.argv[3]) : 2;
```

Comment:
starting with `2` initially, so that:
1. Job scope is small (high chance of success on retry)
2. We prove that chunking N tests per job works as expected.